flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -575,12 +574,10 @@ public void onBuffer(Buffer buffer, int sequenceNumber, int backlog) throws IOEx
 				receivedBuffers.add(buffer);
 				enqueued = true;
 
-				if (listener != null && buffer.isBuffer() && receivedCheckpointId < lastRequestedCheckpointId) {
-					notifyReceivedBuffer = buffer.retainBuffer();
```

Comment:
Even if you go optional, I wouldn't go ternary here. Ternary is meant for some simple condition and one-liners.

Reply:
IMHO ternary is meant for branches that have only one "effect", usually assignment. And this is exactly the case here. 
`If-else` OTH is meant for cases when "effect" is something more complex.

If the condition is complex, it should be extracted to a method, regardless of whether it's `if-else` or ternary operator.

Here, I think the condition is fine (and I didn't change it).

Reply:
As before, it makes the diff/commit/PR more difficult to understand.

Reply:
Reverted to minimize changes in this PR.